packages/slider/src/index.js

```diff
@@ -0,0 +1,638 @@
+        ...(isVertical ? { bottom: handlePosition } : { left: handlePosition }),
+        ...style
+      }}
+      {...dataAttributes}
+      {...props}
+    />
+  );
+});
+
+SliderHandle.displayName = "SliderHandle";
+SliderHandle.propTypes = {};
+
+////////////////////////////////////////////////////////////////////////////////
+export const SliderMarker = forwardRef(function SliderMarker(
+  { children, style = {}, value, ...props },
+  forwardedRef
+) {
+  const {
+    disabled,
+    isVertical,
+    orientation,
+    sliderMin,
+    sliderMax,
+    sliderValue
+  } = useSliderContext();
+
+  const ownRef = useRef(null);
+  const ref = forwardedRef || ownRef;
+  const actualValue = valueToPercent(value, sliderMin, sliderMax);
+  const highlight = sliderValue >= value;
+  const dataAttributes = makeDataAttributes("slider-marker", {
+    orientation,
+    disabled,
+    highlight
+  });
+
+  const absoluteStartPosition = `${actualValue}%`;
+
+  return value != null ? (
+    <div
+      role="presentation"
+      ref={ref}
+      style={{
+        position: "absolute",
+        ...(isVertical
+          ? { bottom: absoluteStartPosition }
+          : { left: absoluteStartPosition }),
+        ...style
+      }}
+      {...dataAttributes}
+      {...props}
+      children={children}
+    />
+  ) : null;
+});
+
+SliderMarker.displayName = "SliderMarker";
+SliderMarker.propTypes = {
+  value: number.isRequired
+};
+
+////////////////////////////////////////////////////////////////////////////////
+export function valueToPercent(value, min, max) {
+  return ((value - min) * 100) / (max - min);
+}
+
+export function percentToValue(percent, min, max) {
+  return (max - min) * percent + min;
+}
+
+export function makeValuePrecise(value, step) {
+  const stepDecimalPart = step.toString().split(".")[1];
+  const stepPrecision = stepDecimalPart ? stepDecimalPart.length : 0;
+  return Number(value.toFixed(stepPrecision));
+}
+
+export function roundValueToStep(value, step) {
+  return makeValuePrecise(Math.round(value / step) * step, step);
+}
+
+export function getAllowedValue(val, min, max) {
+  return val > max ? max : val < min ? min : val;
+}
+
+export function makeDataAttributes(
+  component = "slider",
+  { orientation, highlight, disabled }
+) {
+  return {
+    [`data-reach-${component}`]: "",
+    [`data-reach-${component}-disabled`]: disabled ? "" : undefined,
+    [`data-reach-${component}-orientation`]: orientation,
+    [`data-reach-${component}-highlight`]: highlight ? orientation : undefined
+  };
+}
+
+export function makeId(id, index) {
+  return `${id}:${index}`;
+}
+
+// https://github.com/chakra-ui/chakra-ui/blob/master/packages/chakra-ui/src/utils/index.js#L9
+export function assignRef(ref, value) {
+  if (ref == null) return;
+  if (typeof ref === "function") {
+    ref(value);
+  } else {
+    try {
+      ref.current = value;
+    } catch (error) {
+      throw new Error(`Cannot assign value "${value}" to ref "${ref}"`);
```

Comment:
@eps1lon do you think that it would make sense to bring this variation to Material-UI?

Reply:
> @eps1lon do you think that it would make sense to bring this variation of setRef to Material-UI?

Reply:
At least, it's funny to see Reach using Chakra as inspiration that used Material-UI as inspiration ✌️. @eps1lon, @mbrookes it would probably be great to push https://github.com/mui-org/material-ui/issues/6218 forward.

Reply:
I don't know what this is catching so I can't judge if we need this. Could you open an issue with an example that we don't cover properly?

Reply:
@eps1lon My assumption is that it was added to improve the DX. However, it might be better handled in the prop type.
```js
const refType = PropTypes.oneOfType([PropTypes.func, PropTypes.PropTypes.object]);
```

Ok, let's wait to get a bug report/StackOverflow question that would benefit from it. Thanks for the feedback.

Reply:
I see it as a DX thing since a lot of folks seem to get mixed up when dealing with refs, but no first hand experience with this particular implementation yet. You'll notice that we also use Material as inspiration -- we may or may not end up leaving out the try/catch when I start dropping this hook in its place. https://github.com/reach/reach-ui/blob/1cd033a17746310ef63cdb9ae3dcb10f7e734c17/packages/utils/src/index.js#L83